### ISSUE TITLE
handle better utm params

### ISSUE THREAD

**Issue by tttp** [fixthestatusquo/proca-server#160]
Right now, the utm tracking param definition is:

![image](https://user-images.githubusercontent.com/128426/178486152-b717d39c-9f05-4773-9145-5dd97db82961.png)

this is incoherent as the location is always sent, but not mandatory, while the utm params are mandatory, but not always set, and sometimes the end users do not set all of them

the widget is juggling with a bunch of use case (with or without a location, with or without get utm params, with or without referer, with referers matching known social media). it's heavily dependent on the context and browser and hard to test all the corner cases.

To make it more robust, let's make all the utm params (and location) as NOT mandatory, and add a default value "" for them

rationale: not registering the action because the tracking is somehow incomplete or missing is the worse possible outcome. any default value or discarding the utm is better than rejecting, especially because:
- the error message is hard to parse for the client /not possible to display to the user (too technical)
- the server isn't logging these errors

it seems graphql allows you to specify the default value: 

https://stackoverflow.com/questions/51563960/how-to-add-default-values-to-input-arguments-in-graphql

**Comment by marcinkoziej** [comment]
default values in graphql are a nice find. lets see if absinthe has support

**Comment by marcinkoziej** [comment]
That did not work, but i have changed the requirement to optional, how this does not break any client

**Comment by tttp** [comment]
awesome, thanks and well done!
 did you deploy to our servers already?
